rustworkx-core/src/coloring.rs

```diff
@@ -0,0 +1,144 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use std::cmp::Reverse;
+use std::hash::Hash;
+
+use crate::dictmap::*;
+use hashbrown::{HashMap, HashSet};
+use petgraph::visit::{EdgeRef, IntoEdges, IntoNodeIdentifiers, NodeCount};
+use rayon::prelude::*;
+
+/// Color a graph using a greedy graph coloring algorithm.
+///
+/// This function uses a `largest-first` strategy as described in [1]_ and colors
```

Comment:
I don't think this footnote format works in rustdoc. Looking at the CI run of `cargo doc` it looks like it wasn't happy with this either:

https://github.com/Qiskit/rustworkx/actions/runs/5010849131/jobs/8981094523#step:14:46

Looking at the rustdoc documentation the syntax for footnotes is: https://doc.rust-lang.org/stable/rustdoc/how-to-write-documentation.html?highlight=footnote#footnotes but it might make sense to just put it inline instead of using a footnote here.

Reply:
I pushed up https://github.com/Qiskit/rustworkx/pull/875/commits/4cd566503af77ba7bea56b1c28c090cd5e4bbec0 to resolve this.

Reply:
@mtreinish, thanks for taking care of this.